spyder/plugins/statusbar/plugin.py

```diff
@@ -93,15 +102,28 @@ def add_status_widget(self, widget, index=0):
             )
 
         # Check it was not added before
-        if id_ in self.STATUS_WIDGETS and not running_under_pytest():
+        in_widgets = (id_ in self.STATUS_WIDGETS or
+                      id_ in self.EXTERNAL_LEFT_WIDGETS or
+                      id_ in self.EXTERNAL_LEFT_WIDGETS or
+                      id_ in self.INTERNAL_WIDGETS)
+        if in_widgets and not running_under_pytest():

```

Comment:
I think we can restore the previous line here because `STATUS_WIDGETS` contains all status bar widgets. In other words, my suggestion is to leave

```python
if id_ in self.STATUS_WIDGETS and not running_under_pytest():
```

instead of these four new lines.